src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -562,17 +572,27 @@ export async function refundRequestIsValid(
     return { valid: false, reason: "Unable to find matching deposit" };
   }
 
-  // Verify that the refundToken maps to a known HubPool token.
+  // Verify that the refundToken maps to a known HubPool token and is the correct
+  // token for the chain where the refund was sent from.
   // Note: the refundToken must be valid at the time of the Fill *and* the RefundRequest.
-  // @todo: Resolve to the HubPool block number at the time of the RefundRequest ?
-  const hubPoolBlockNumber = hubPoolClient.latestBlockNumber ?? hubPoolClient.deploymentBlock - 1;
+  // @todo: This really should be using the hub pool block equivalent of the fillBlock. Its unlikely to be a problem
+  // since L1 tokens are rarely expected to be remapped but its worth solving.
+  const hubPoolBlockNumber = hubPoolClient.latestBlockNumber!;
```

Comment:
This is an annoying artifact. We really should be passing in a mainnet bundle end block, but that would require adding an optional param to this call (and all the way up the stack).

Its pretty ugly but we might have to do it to be correct